app/models/provisioning_template.rb

```diff
@@ -4,7 +4,17 @@ class ProvisioningTemplate < Template
   friendly_id :name
   include Parameterizable::ByIdName
 
+  class << self
```

Comment:
Not against this by any chance - it's an ugly feature. I wonder though, isn't `Template` an abstract class effectively? As far as I can see it's only instantiated in tests, and according to the link in here,  `self.abstract_class = true` on `Template` would solve the problem too, with less repetition & preventing the class from being instantiated.

Reply:
I'll try that, it would make sense in case it works. I'll let you know how it goes.

Reply:
Sadly it does not work, it fails on finding the right table name
```
 | ActiveRecord::StatementInvalid (PG::UndefinedTable: ERROR:  relation "provisioning_templates" does not exist
 | LINE 5:                WHERE a.attrelid = '"provisioning_templates"'...
 |                                           ^
 | :               SELECT a.attname, format_type(a.atttypid, a.atttypmod),
 |                      pg_get_expr(d.adbin, d.adrelid), a.attnotnull, a.atttypid, a.atttypmod
 |                 FROM pg_attribute a LEFT JOIN pg_attrdef d
 |                   ON a.attrelid = d.adrelid AND a.attnum = d.adnum
 |                WHERE a.attrelid = '"provisioning_templates"'::regclass
 |                  AND a.attnum > 0 AND NOT a.attisdropped
 |                ORDER BY a.attnum
 | ):
 |   app/controllers/application_controller.rb:231:in `resource_base_with_search'
 |   app/controllers/application_controller.rb:235:in `resource_base_search_and_page'
 |   app/controllers/templates_controller.rb:14:in `index'
 |   app/controllers/concerns/application_shared.rb:15:in `set_timezone'
 |   app/models/concerns/foreman/thread_session.rb:32:in `clear_thread'
 |   app/controllers/concerns/foreman/controller/topbar_sweeper.rb:12:in `set_topbar_sweeper_controller'
 |   lib/middleware/catch_json_parse_errors.rb:8:in `call'
 |   lib/middleware/session_safe_logging.rb:17:in `call'
 |   lib/middleware/tagged_logging.rb:18:in `call'
 | 
```

if I left only `self.table_name` calls it failed to load associations `Association named 'template_kind' was not found on Ptable; perhaps you misspelled it?`, IIRC the we had the same issue parameters STI, which is why we moved them to parent class. I'd like to avoid this here.